### PR TITLE
feat(compute): add compute disk create secondary custom sample

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description
Documentation - https://cloud.google.com/compute/docs/disks/async-pd/configure?authuser=0#gcloud_1
Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateDiskSecondaryCustom {
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java:34]: Considering refactoring the class name to `CreateSecondaryCustomDisk`.

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+  // Creates a secondary disk with specified custom parameters.
+  public static Disk createDiskSecondaryCustom(String primaryProjectId, String secondaryProjectId,
+      String primaryDiskName, String secondaryDiskName, String primaryDiskZone,
+      String secondaryDiskZone, long diskSizeGb,  String diskType)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    String primaryDiskSource = String.format("projects/%s/zones/%s/disks/%s",
+        primaryProjectId, primaryDiskZone, primaryDiskName);
+
+    DiskAsyncReplication asyncReplication = DiskAsyncReplication.newBuilder()
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java:72]: Please move all compute API related code within the try catch. Applies to all samples across PRs.

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          .putAllLabels(labels)
+          .setAsyncPrimaryDisk(asyncReplication)
+          .build();
+
+      // Wait for the create disk operation to complete.
+      Operation response = disksClient.insertAsync(secondaryProjectId, secondaryDiskZone, disk)
+          .get(3, TimeUnit.MINUTES);
+
+      if (response.hasError()) {
+        return null;
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java:105]: Print and then throw an error.

**Review by Sita04 — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java (hunk 3 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java:105]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java (hunk 1 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateDiskSecondaryCustom.java:34]: Thank you! Renamed

File: compute/cloud-client/src/main/java/compute/disks/CreateSecondaryCustomDisk.java [hunk 4]
```diff
@@ -102,9 +102,9 @@ public static Disk createDiskSecondaryCustom(String primaryProjectId, String sec
           .get(3, TimeUnit.MINUTES);
 
       if (response.hasError()) {
-        return null;
+        throw new Error("Error creating secondary custom disks! " + response.getError());
       }
-      return disksClient.get(secondaryProjectId, secondaryDiskZone, secondaryDiskName);
+      return response.getStatus();
```
Comment (Sita04) [inline compute/cloud-client/src/main/java/compute/disks/CreateSecondaryCustomDisk.java:107]: Return the operation response.

**Review by Sita04 — APPROVED** [review]
Kindly address the comment before merging.

File: compute/cloud-client/src/main/java/compute/disks/CreateSecondaryCustomDisk.java (hunk 4 above)
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/CreateSecondaryCustomDisk.java:107]: Returned Operation.Status as was done for other samples

**Pull request merged**